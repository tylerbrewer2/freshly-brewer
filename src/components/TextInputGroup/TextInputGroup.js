import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// components
import Label from '../Label';
import TextInput from '../TextInput';
import Col from '../Col';

const TextInputGroupCore = ({
    className,
    labelText,
    value,
    onChange,
    identifier,
    required,
    width
  }) => (
  <Col width={width}>
    <div className={className}>
      <Label>{labelText}</Label>
      <TextInput
        value={value}
        onChange={onChange}
        identifier={identifier}
      />
    </div>
  </Col>
)

TextInputGroupCore.propTypes = {
  labelText: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  identifier: PropTypes.string,
  required: PropTypes.bool,
  width: PropTypes.number,
}

TextInputGroupCore.defaultProps = {
  identifier: null,
  required: false,
  width: 12,
}

const TextInputGroup = styled(TextInputGroupCore)`
margin-bottom: 30px;
`

export default TextInputGroup;