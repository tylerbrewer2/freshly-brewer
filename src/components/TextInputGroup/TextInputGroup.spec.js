import React from 'react';
import renderer from 'react-test-renderer';

import TextInputGroup from './TextInputGroup';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextInputGroup
        identifier="identifier"
        value="test value"
        onChange={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
