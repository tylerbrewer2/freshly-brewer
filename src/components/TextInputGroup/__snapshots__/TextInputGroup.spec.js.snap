exports[`component structure renders correctly 1`] = `
<div
  className="col-xs-12 
      undefined 
      undefined">
  <div
    className="sc-htpNat hvvDZo">
    <label
      className="sc-bdVaJa fxdtZh"
      htmlFor="identifier" />
    <input
      className="sc-bwzfXH gmTCEv"
      id="identifier"
      name="identifier"
      onChange={[Function]}
      required={false}
      type="text"
      value="test value" />
  </div>
</div>
`;
